File: python/inflearn/LinkedList.py
class Node:
	def __init__(self, value = 0, next = None):
		self.value = value
		self.next = next

class LinkedList(object):
  def __init__(self):
    self.head = None
    self.tail = None

  def append(self, value):
    new_node = Node(value)
    if self.head is None:
      self.head = new_node
      self.tail = new_node
    else:
      self.tail.next = new_node
      self.tail = self.tail.next

  def get(self, idx):
    current = self.head
    for _ in range(idx):
      current = current.next
    return current.value

  def insert(self, idx, value):
    new_node = Node(value)
    current = self.head
    for _ in range(idx - 1):
      current = current.next
    new_node.next = current.next
    current.next = new_node

  def remove(self, idx):
    current = self.head
    for _ in range(idx - 1):
      current = current.next
    current.next = current.next.next


li = LinkedList()
li.append(1)
li.append(2)
li.append(3)
li.append(4)
li.get(0)
li.get(1)
li.get(3)
li.insert(idx = 2, value = 9)

File: javascript/etc/LinkedList.js
class Node {
  constructor(value = 0, next = null) {
    this.value = value;
    this.next = next;
  }
}

class LinkedList {
  constructor() {
    this.head = null;
    this.tail = null;
  }

  append(value) {
    let newNode = new Node(value);
    if(this.head === null) {
      this.head = newNode;
      this.tail = newNode;
    } else {
      this.tail.next = newNode;
      this.tail = this.tail.next;
    }
  }

  get(idx) {
    let current = this.head;

    for (let i = 0; i < idx; i++) {
      current = current.next;
    }
    console.log(current.value);
    return current.value;
  }

  insertAt(idx, value) {
    let current = this.head;
    let newNode = new Node(value);

    for (let i = 0; i < idx - 1; i++) {
      current = current.next;
    }

    newNode.next = current.next;
    current.next = newNode;
  }

  removeAt(idx) {
    let current = this.head;

    for (let i = 0; i < idx - 1; i++) {
      current = current.next;
    }
    current.next = current.next.next;
  }


}

let ll = new LinkedList();
ll.append(1);
ll.append(2);
ll.append(3);
ll.append(4);
ll.insertAt(2, 11)
ll.removeAt(3)
ll.get(0);
ll.get(1);
ll.get(2);
ll.get(3);